src/zimg/colorspace/operation_impl.cpp

```diff
@@ -69,6 +69,78 @@ class GammaOperationC final : public Operation {
 	}
 };
 
+class AribB67OperationC final : public Operation {
+	float m_kr;
+	float m_kg;
+	float m_kb;
+	float m_scale;
+public:
+	AribB67OperationC(double kr, double kg, double kb, float scale) :
+		m_kr{ static_cast<float>(kr) },
+		m_kg{ static_cast<float>(kg) },
+		m_kb{ static_cast<float>(kb) },
+		m_scale{ scale }
+	{}
+
+	void process(const float * const *src, float * const *dst, unsigned left, unsigned right) const override
+	{
+		const float gamma = 1.2f;
+
+		for (unsigned i = left; i < right; ++i) {
+			float r = src[0][i] * m_scale;
+			float g = src[1][i] * m_scale;
+			float b = src[2][i] * m_scale;
+
+			float yd = std::max(m_kr * r + m_kg * g + m_kb * b, FLT_MIN);
+			float ys = zimg_x_powf(yd, (gamma - 1.0f) / gamma);
```

Comment:
z.lib uses 0-1 range for gamma-encoded domain, which makes the 'alpha' factor unity according to BT.2100.